### DISCUSSION TITLE
The opposite of ".on()"?

### DISCUSSION THREAD

**Discussion by alexandernst** [clientIO/joint#2143] [Q&A] [answered]
### Introduction

I have a function that is triggered when a link's source is changed:

```js
class MyLink {
  constructor(...) {
    super(...);

    ...

    this.on("change:source", foo);
  }

  foo() {
    console.log("foo!");
  }
}
```

Having a listener for each link that already has a `source` is not good for the performance (and useless in my case, since in my use case links can't change their `source` once they have been created). Is there a way I can remove a listener from a link once the event has been triggered? Something like:

```js
this.on("change:source", foo);

foo() {
  console.log("foo!");
  this.off("change:source", foo);
}
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
Yes, there is [off()](https://backbonejs.org/#Events-off) method too.

**Reply by alexandernst** [reply]
Oh, I don't know how I missed it! Thank you :)
